sources/refund.move

```diff
@@ -117,6 +129,71 @@ module refund::refund {
         vector::destroy_empty(amounts);
     }
 
+    // === Phase 1: Timeout setup ===
+
+    public entry fun set_timeout_ts(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        timeout_ts: u64,
+        clock: &Clock,
+    ) {
+        assert_timeout_setup_phase(pool);
+        assert_publisher(pub);
+        assert!(timeout_ts > clock::timestamp_ms(clock), EInvalidTimeoutTimestamp);
+
+        option::fill(&mut pool.timeout_ts, timeout_ts);
+    }
+
+    // === Phase 2: Funding ===
+
+    /// Permissionless endpoint for funding the fund pool with the given SUI coin.
+    public entry fun fund(
+        pool: &mut RefundPool,
+        coin: Coin<SUI>,
+        ctx: &mut TxContext,
+    ) {
+        assert_funding_phase(pool);
+
+        let amount = coin::value(&coin);
+        refund_table::insert_or_add(funders_mut(&mut pool.base_pool), sender(ctx), amount);
+
+        // If overfunded then return the excess
+        let total_to_refund = total_to_refund(&pool.accounting);
+        let total_raised = total_raised_mut(&mut pool.accounting);
+
+        let is_overfunded = *total_raised + amount > total_to_refund;
+        if (is_overfunded) {
```

Comment:
For simplicity, I would just assert `!is_overfunded` and make it the caller's problem to pass a coin that does not push the amount raised over the threshold.

Reply:
Implemeted